src/main/resources/js/asmProvidedLibs.js

```diff
@@ -1,554 +1,423 @@
-        }
-        _.indentLevel++;
-    };
-
-
-    console.groupCollapsed = function groupCollapsed(args) {
-        console.group.apply(null, _.slice(arguments, 0));
-    };
-
-
-    /**
-     * Closes the most recently opened block created by a call
-     * to 'console.group()' or 'console.groupCollapsed()'.
-     */
-    console.groupEnd = function groupEnd() {
-        if (_.indentLevel < 1) return;
-        _.indentLevel--;
-    };
-
-
-    /**
-     * Logs a message with the visual "info" representation.
-     */
-    console.info = function info(args) {
-        const s = _.formatArguments(arguments);
-        _.writeln("info", s);
-    };
-
-
-    /**
-     * Logs a message with the visual "log" representation.
-     */
-    console.log = function log(args) {
-        const s = _.formatArguments(arguments);
-        _.writeln("log", s);
-    };
-
-
-    console.time = function time(label) {
-        if (typeof label === "undefined") label = "default";
-        label = String(label);
-        if (label in _.timers) return;
-        _.timers[label] = Date.now();
-    };
-
-
-    /**
-     * Stops a timer created by a call to `console.time(label)` and logs the elapsed time.
-     */
-    console.timeEnd = function timeEnd(label) {
-        if (typeof label === "undefined") label = "default";
-        label = String(label);
-        const milliseconds = Date.now() - _.timers[label];
-        _.writeln("timeEnd", label + ": " + milliseconds + " ms");
-    };
-
-
-    /**
-     * Logs a stack trace for where the call occurred from, using the given arguments as a label.
-     */
-    console.trace = function trace(args) {
-        const label = "Trace" + ("0" in arguments ? ": " + _.formatArguments(arguments) : "");
-        const e = new Error();
-        Error.captureStackTrace(e, trace);
-        // Replaces the first line by our label.
-        const s = label + "\n" + e.stack;
-        _.writeln("trace", s);
-    };
-
-
-    /**
-     * @todo Logs a tabular representation of the given data.
-     * Fall back to just logging the argument if it can’t be parsed as tabular.
-     */
-    console.table = function table(tabularData, properties) {
-        console.log(tabularData);
-    };
-
-
-    /**
-     * Logs a message with the visual "warning" representation.
-     */
-    console.warn = function warn(args) {
-        const s = _.formatArguments(arguments);
-        _.writeln("warn", s);
-    };
-
-    return console;
+let global = this;
+
+global.Java = {
+    type: function (clazz) {
+        const split = clazz.split(".");
+
+        let returned = Packages;
+
+        for (let i = 0; i < split.length; i++) {
+            returned = returned[split[i]];
+        }
+
+        return returned;
+    }
+};
+
+global.sync = (func, lock) => new org.mozilla.javascript.Synchronizer(func, lock);
+
+// Thread
+global.Thread = Java.type("com.chattriggers.ctjs.minecraft.wrappers.objects.threading.WrappedThread");

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['Java' is not defined.](https://app.codacy.com/manual/FalseHonesty/ct.js/pullRequest?prid=5138651)